akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -317,6 +317,22 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Limit[T](n: Int) extends PushStage[T, T] {
+  private var left = n
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
+    left -= 1
+    if (left >= 0) ctx.push(elem)
+    else {
+      if (ctx.isFinishing) ctx.push(elem)
```

Comment:
Yes, you are correct. I'm just about to push the change.

Also, when `n < 0`, currently I'm following the convention of adding a `require(n >= 0, ...)` on `akka.stream.impl.Limit` class so that it throws an `IllegalArgumentException`. But we can also terminate the stream with `StreamLimitReachedException` instead